JOING-34 feat: 크리에이터 프로필 유해성 평가 API 구현

hyoom opened the pull request:
## #️⃣ 연관된 이슈

> #10  #JOING-34

<br/>

## 📝 작업 내용

> 크리에이터 프로필 유해성 평가 기능 구현하였습니다.
> 클라이언트에서 백으로 요청이 들어오면, 백에서 AI 서버에 요청하여 해당 크리에이터의 채널에 대한 유해성 평가 결과를 가지고 옵니다.

<br/>

## 📷 스크린샷(선택)

> 이미지가 있다면 첨부해주세요

<br/>

## 💬 리뷰 요구사항(선택)

> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
>
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?
>

hyoom committed:
JOING-34 fix: UserService통해 channelId 저장하도록 추가

hyoom committed:
JOING-34 feat: 크리에이터 프로필 유해성 평가 기능 구현

hyoom committed:
JOING-34 refactor: 응답 DTO 구조 변경

- 불필요한 필드 제거 (score, createdDateTime, updatedDateTime)
- 새로운 응답 필드 추가 (email, summaryTitle, SummaryContent, SummaryKeyword)

jayden-bin commented on src/main/java/com/ktb/joing/domain/item/dto/response/ItemDetailResponse.java:
```diff
@@ -0,0 +1,47 @@
+                Arrays.asList(item.getSummary().getKeyword().split(","))
+        );
+    }
+}
```
<img width="147" alt="스크린샷 2024-11-29 오전 10 20 57" src="https://github.com/user-attachments/assets/8e684cf1-9f64-49ba-8cc6-39db58669890">
파일끝 개행 설정을 IDE에 해주시면 좋을것 같아요~!

hyoom replied:
넵 수정했습니다!

jayden-bin commented on src/main/java/com/ktb/joing/domain/user/service/UserService.java:
```diff
@@ -84,4 +98,26 @@ private void validateDuplicateNickname(String nickname) {
+
+        ProfileEvaluationRequest request = creatorEvaluationRequest(creator);
+
+        return reactiveHttpService.post(
+                aiUrl + "/ai/profile/evaluation",
+                request,
+                ProfileEvaluationResponse.class
+        )
+        .doOnError(e -> log.error("Profile evaluation failed: {}", e.getMessage()))
+        .onErrorMap(e -> new UserException(UserErrorCode.PROFILE_EVALUATION_FAILED));
+    }
```
레이어드 아키텍쳐를 사용한다면 요런 API만 요청하는 레이어를 별도로 두어도 괜찮을거예요!
저희 회사에서는 API요청하는 레이어를 ~~~Client라고 부르고 해당 레이어는 순수하게 API요청만 담당해요

hyoom replied:
API 요청만 딱 전담하게 하면 역할이 확실히 나뉘어서 관리하기도 훨씬 편하고 좋은것 같아요! 좋은 아이디어 공유해주셔서 감사합니다 ~

jayden-bin commented on src/main/java/com/ktb/joing/domain/user/dto/request/ProfileEvaluationRequest.java:
```diff
@@ -0,0 +1,17 @@
+    private String channelId;
+
+    @Builder
+    public ProfileEvaluationRequest(String channelId) {
+        this.channelId = channelId;
+    }
```
으흠 ㅋㅋㅋ 얘는 프로퍼티가 단 한개인데 빌더패턴으로 쓰는게 맞을지는 약간 의문이 드네요

hyoom replied:
ProfileEvaluationRequest request = new ProfileEvaluationRequest(creator.getChannelId());로 변경해서 간단하게 처리하도록 수정했습니다!

hyoom committed:
Merge branch 'develop' into feature/JOING-34-profile-evaluation

hyoom committed:
JOING-34 chore: 파일끝 개행 설정

hyoom committed:
JOING-34 feat: AI API요청만 담당하는 로직을 Client 레이어로 분리

- item 도메인에 ItemAIClient 추가
- user 도메인에 ProfileAIClient 추가
- 기존 Service 레이어의 API 호출 로직을 Client로 이동

hyoom committed:
JOING-34 refactor: ProfileEvaluationRequest에서 불필요한 빌더 패턴 제거

hyoom committed:
JOING-34 refactor: 기획안 단건 조회 응답 DTO 구조 변경
